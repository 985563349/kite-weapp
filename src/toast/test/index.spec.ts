import path from 'node:path';
import simulate from 'miniprogram-simulate';

describe('toast', () => {
  const kToast = simulate.load(path.resolve(__dirname, '../toast'), 'k-toast', {
    rootPath: path.resolve(__dirname, '../../'),
  });

  test('should render toast add match snapshot', async () => {
    const comp = simulate.render(
      simulate.load({
        usingComponents: {
          'k-toast': kToast,
        },
        template: `<k-toast visible />`,
      })
    );
    comp.attach(document.createElement('parent-wrapper'));

    await simulate.sleep(16 * 3); // wait three frames for the animation to complete.
    expect(comp.toJSON()).toMatchSnapshot();
  });
});
