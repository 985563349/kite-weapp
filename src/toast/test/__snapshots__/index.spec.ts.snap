// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toast should render toast add match snapshot 1`] = `
<main>
  <wx-view
    class="k-toast k-fade-enter-to k-fade-enter-active enter-to-class enter-active-class k-class"
    id=""
    style="-webkit-transition-duration:300ms;transition-duration:300ms;z-index:11000"
    catch:touchmove="noop"
    bind:transitionend="onTransitionEnd"
  />
</main>
`;
